### PR TITLE
Add integration tests for logic triggered by websockets

### LINKED ISSUE #127: Websocket integration tests
Write an integration test suite to verify that the correct state is set, correct events are emitted, transactionQueue is updated etc based on possible websocket initialisation, and websocket messages received from the server

### PR DESCRIPTION
**Pull request by liamaharon**
Closes #127 

TODO:
- [x] Decide where to handle socket connection retries

New tests: 
Screenshot from 2019-04-30 13-05-44

### CODE REVIEW

**Comment by liamaharon** [comment]
@cmeisl the library only retries creating websocket connections when calling `logEvent` in `events.js`.

If we choose to leave the code as-is I think it would be best to test the retries in the same suite as the tests covering `events.js` which contains `logEvent` (will get around to this probably this week) rather than in the suite covering websocket functionality.

Another option would be to move the retry logic into `websockets.js`. This would cause all calls to open the socket trigger retries if they're not successful. 

It'll be less work to leave the retry logic in `logEvent` but may make more sense for the functionality to be handled in the `websockets.js` file. Let us know what you think.

**Comment by cmeisl** [comment]
I think retry belongs in `websockets.js`. That seems like the proper separation of concerns.

File: src/js/index.js [hunk 1]
```diff
@@ -456,7 +456,7 @@ function init(config) {
 
 // GETSTATE FUNCTION //
 
-function getState() {
+export function getState() {
```
Comment (cmeisl) [inline src/js/index.js]: @liamaharon What is the reason for this change?

Comment (liamaharon) [reply src/js/index.js]: As mentioned in Slack this is redundant, my mistake for leaving it in. Good pickup.
